generic_server.cc

```diff
@@ -115,32 +113,31 @@ server::~server()
 }
 
 future<> server::stop() {
-    if (!_stopping) {
-        co_await shutdown();
-    }
-
-    co_await _all_connections_stopped.get_future();
+    co_await shutdown();
+    co_await std::move(_all_connections_stopped);
```

Comment:
better `co_await std::exchange(_all_connections_stopped, make_ready_future<>())`

Reply:
IIUC, this is to make `server:stop()` idempotent as well. Will do it in v6, but I think it deserves its own patch.